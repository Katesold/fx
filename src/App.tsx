import GlobalStyle from "./components/global";
import { AppContainer } from './AppContainer';
import { Button } from './components/Button/Button';
import { Table } from "./components/Table/Table";
function App() {
  return (
    <>
      <GlobalStyle />
      <AppContainer>
        <Table data={[{
              name: 'test',
              id: 'test1',
              amount: 10,
              currency: 'Â£',
              reference: '124',
        }]}/>
        <Button text="Button Text" />
      </AppContainer>
    </>
  );
}

export default App;
