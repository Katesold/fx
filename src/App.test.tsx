import { act, fireEvent, render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import App from './App';

// class FileReaderMock {
//   DONE = FileReader.DONE;
//   EMPTY = FileReader.EMPTY;
//   LOADING = FileReader.LOADING;
//   readyState = 0;
//   error: FileReader['error'] = null;
//   result: FileReader['result'] = null;
//   abort = jest.fn();
//   addEventListener = jest.fn();
//   dispatchEvent = jest.fn();
//   onabort = jest.fn();
//   onerror = jest.fn();
//   onload = jest.fn();
//   onloadend = jest.fn();
//   onloadprogress = jest.fn();
//   onloadstart = jest.fn();
//   onprogress = jest.fn();
//   readAsArrayBuffer = jest.fn();
//   readAsBinaryString = jest.fn();
//   readAsDataURL = jest.fn();
//   readAsText = jest.fn();
//   removeEventListener = jest.fn();
// }

describe('App components are being tested', () => {
  const file = new File(['1, 34, 342, 25'], 'file.csv', {type: 'text/csv'});
  // const fileReader = new FileReaderMock();
  // jest.spyOn(window, 'FileReader').mockImplementation(() => fileReader);

  it('should match snapshot', () => {
    const { container } = render(<App />);
    expect(container).toMatchSnapshot();
  });

  it('should display UploadContainer', () => {
    render(<App />);
    const title = screen.getByText('Upload Payments');
    expect(title).toBeInTheDocument();
    expect(screen.getByRole('presentation')).toBeInTheDocument();
  });

  it('should render Toast', () => {
    render(<App />);
    const toast = document.querySelector('.Toastify');
    expect(toast).toBeInTheDocument();
  });

  it('should accept file', async () => {
    render(<App />);
    // const file = new File(["(⌐□_□)"], "chucknorris.png", { type: "image/png" });
    // const file = new File(["(⌐□_□)"], "test.csv", { type: "text/csv" });

    // fileReader.result = 'file content, two';
    // fileReader.addEventListener.mockImplementation((_, fn) => fn());

    // const content = await fileReader.readAsText(file);

    // expect(content).toBe('file content');

    const inputEl = document.querySelector('input') as HTMLInputElement;
    await waitFor(() => {
      userEvent.upload(inputEl, file);
    });
    expect(inputEl.files).toHaveLength(1);
  });
});

