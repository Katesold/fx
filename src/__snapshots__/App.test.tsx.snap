// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App components are being tested should match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe chvWhX"
  >
    <div
      class="sc-kDvujY klcgqA"
    >
      <h1>
        Upload Payments
      </h1>
      <div
        class="sc-ipEyDJ gwRKMM dropzone"
        role="presentation"
        tabindex="0"
      >
        <input
          accept="text/csv,.csv,.xls,.xlsx"
          multiple=""
          style="display: none;"
          tabindex="-1"
          type="file"
        />
        <img
          alt="cloud"
          src="cloud.png"
        />
        <h4>
          Drag & Drop
        </h4>
        <p
          class="sc-csuSiG hNyWrk"
        >
          Supported file types: 
          <span>
            .csv .xls .xlsx
          </span>
        </p>
      </div>
      <div
        class="Toastify"
      />
    </div>
  </div>
</div>
`;
